packages/synapse-react-client/src/components/AccessRequirementList/AccessRequirementList.tsx

```diff
@@ -44,20 +28,48 @@ import {
 } from '../../synapse-queries'
 import TwoFactorAuthEnabledRequirement from './RequirementItem/TwoFactorAuthEnabledRequirement'
 import { AccessRequirementListItem } from './AccessRequirementListItem'
-import { useSynapseContext } from '../../utils/context/SynapseContext'
+import { useSynapseContext } from '../../utils'
 import { useCanShowManagedACTWikiInWizard } from './AccessRequirementListUtils'
-import { DialogActions } from '@mui/material'
+import { noop } from 'lodash-es'
+import { DialogBase } from '../DialogBase'
+import UserOrTeamBadge from '../UserOrTeamBadge'
+import { EntityLink } from '../EntityLink'
 
 export type AccessRequirementListProps = {
```

Comment:
Summary of changes to the props:

- Caller must specify one of (subjectId AND subjectType), entityId, teamId
- entityId, teamId are deprecated
- Made `onHide` optional, since it only make sense when `renderAsModal` is true